execution/execution.go

```diff
@@ -11,10 +11,13 @@ import (
 	"github.com/pkg/errors"
 )
 
+type Task func() error
+
 // This struct represents additional metadata to be returned with Get() and used recursively (like WatermarkSource)
 type ExecutionOutput struct {
```

Comment:
actually, after submitting approve, I would like You to write proper comments to `e v e r y` single field of ExecutionOutput as it grows!

Reply:
sure